include/boost/geometry/algorithms/detail/buffer/turn_in_piece_visitor.hpp

```diff
@@ -180,7 +180,12 @@ class turn_in_piece_visitor
             // unit tests of hard cases start to fail (5 in multi_polygon)
             // But it is acknowlegded that such a threshold depends on the
             // scale of the input.
+#if defined(BOOST_GEOMETRY_USE_RESCALING)
             if (state.m_min_distance > 1.0e-4 || ! state.m_close_to_offset)
+#else
+            constexpr double zero = 0;
```

Comment:
Shouldn't this be of the same type as `state.m_min_distance`?

Reply:
Indeed. I'm considering an alternative fix (that takes a bit longer), but otherwise I'll change it.